pkg/logql/syntax/ast.go

```diff
@@ -1101,9 +1101,13 @@ type Grouping struct {
 // impls Stringer
 func (g Grouping) String() string {
 	var sb strings.Builder
+	if g.Groups == nil {
+		return ""
+	}
```

Comment:
Sorry, maybe it was not clear in my previous [comment](https://github.com/grafana/loki/pull/9417#pullrequestreview-1418427105), but what states that an expression does not have any grouping is to set `Grouping` to nil. Therefore, in those cases, the `String()` method of the Grouping struct should not even happen.

Reply:
Additionally, if `g.Groups` is `nil` it should be the same as if `len(g.Groups) == 0`

Reply:
@ssncferreira An empty Grouping struct is added to VectorAggregationExpr [here](https://raintank-corp.slack.com/archives/C029V4SSS9L/p1684769654536579). That causes a `by ()` clause to be added to every `avg`, `sum` and so on. Is this expected?